exchange_machine.lua

```diff
@@ -531,39 +495,41 @@ local function handle_own_orders(player, formname, fields)
 		show_own_orders(p_name, state.own_results)
 	end
 
-	if fields["result_table"] then
-		local event = minetest.explode_table_event(fields["result_table"])
+	if fields.result_table then
+		local event = minetest.explode_table_event(fields.result_table)
 		if event.type == "CHG" then
 			state.selected_index = event.row - 1
 			show_own_orders(p_name, results, event.row)
 		end
 	end
 
-	if fields["back"] then
+	if fields.back then
 		show_main(p_name)
 	end
-
-	return true
 end
 
 
-local function handle_summary(player, formname, fields)
-	if formname ~= summary_form then return end
-
-	local p_name = player:get_player_name()
-
-	if fields["back"] then
-		show_main(p_name)
+local function handle_summary(player, fields)
+	if fields.back then
+		show_main(player:get_player_name())
 	end
-
-	return true
 end
 
 
-minetest.register_on_player_receive_fields(handle_main)
-minetest.register_on_player_receive_fields(handle_select)
-minetest.register_on_player_receive_fields(handle_own_orders)
-minetest.register_on_player_receive_fields(handle_summary)
+minetest.register_on_player_receive_fields(function(player, formname, fields)
```

Comment:
If this is an optimization it is probably not that significant, since fields are rarely received and three function calls/iterations through the callback table are the least of the cost